source/state_representation/include/state_representation/Robot/Jacobian.hpp

```diff
@@ -159,28 +173,35 @@ class Jacobian : public State {
    * the name of the output CartesianTwist will be "robot"_end_effector and
```

Comment:
Is this docstring correct? I don't see where the names are set to end effector and base

Reply:
that is a very good catch. It used to be but this was lost in refactoring. Thinking about it I am not entirely sure this is the bast behavior but I don't really see how to make it differently. I will add it back we can think about it at a later stage

Reply:
actually, the jacobian could have a reference frame and a frame name as well. I think it would make a lot of sense. Then this would come from there.

Reply:
so it seems it was still set and not lost (line 86 in `Jacobian.cpp`):

```
CartesianTwist result(this->get_name() + "_end_effector", twist, this->get_name() + "_base");
```

but this is definitely something to look into so I will make an issue as a reminder

Reply:
see issue #29